### PR TITLE
Refactor store for teach page 

### PR DESCRIPTION
**Pull request by y4izus**
## Changes

Part of https://github.com/Qiskit/saiba/issues/351

Fetches the admin courses for the teach page directly from the service

## Implementation details

We don't need to have the admin courses on the store because they are only used on the teach page, so this PR fetches the data directly from the services and remove the unneeded code

## How to read this PR

Go to `/account/teach` and check if everything is working as expected
Also, check the code to see if I'm missing something or if we can improve anything

### CODE REVIEW

**Review by Tansito — DISMISSED** [review]
Thanks for this @y4izus ! With the changes I think the flow is quite simpler to follow than before 🎉

File: components/teach/Courses.vue [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
 
 import '@carbon/web-components/es/components/skeleton-text/index.js';
 
-const courseStore = useCourseStore();
 const userStore = useUserStore();
-const { adminCourses: courses } = storeToRefs(courseStore);
 const { currentUser } = storeToRefs(userStore);
 
-const { pending } = await courseStore.fetchAdminCourseData(
-  // The page using this component requires authentication
```
Comment (abdonrd) [inline components/teach/Courses.vue:21]: I think we can keep this comment to explain the `!` in the `currentUser.value!.id`.

Comment (abdonrd) [inline components/teach/Courses.vue:21]: I think we can keep this comment to explain the `!` in the `currentUser.value!.id`.

**Review by abdonrd — APPROVED** [review]
Looks good, thanks Yaiza!

**Review by abdonrd — APPROVED** [review]

**Pull request merged**